src/main/utils/ipcComms.ts

```diff
@@ -7,7 +7,7 @@ export default function setupIpcComs(
     tray: Tray,
     mainWindow: BrowserWindow
 ): void {
-    ipcMain.on('select-dir', async (event) => {
+    ipcMain.handle('select-dir', async () => {
         const dialogWindow = new BrowserWindow({
```

Comment:
if didn't know  we can open a dialog without passing a window , we should do that here too